### PR TITLE
Extract and improve test macros

### PR DESCRIPTION
**Pull request by grigoryk**
This folds together relevant changes from https://github.com/ncalexan/mentat/commits/rewind and #632.

### CODE REVIEW

File: db/src/debug.rs [hunk 1]
```diff
@@ -252,3 +308,168 @@ pub(crate) fn dump_sql_query(conn: &rusqlite::Connection, sql: &str, params: &[&
     let dump = String::from_utf8(tw.into_inner().unwrap()).unwrap();
     Ok(dump)
 }
+
+
+// Macro to parse a `Borrow<str>` to an `edn::Value` and assert the given `edn::Value` `matches`
```
Comment (ncalexan) [inline db/src/debug.rs]: nit: kill one of these blank lines.

File: db/src/debug.rs [hunk 2]
```diff
@@ -252,3 +308,168 @@ pub(crate) fn dump_sql_query(conn: &rusqlite::Connection, sql: &str, params: &[&
...
+
+    pub fn last_tx_id(&self) -> Entid {
+        self.partition_map.get(&":db.part/tx".to_string()).unwrap().index - 1
+    }
+
+    pub fn last_transaction(&self) -> edn::Value {
+        transactions_after(&self.sqlite, &self.schema, self.last_tx_id() - 1).expect("last_transaction").0[0].into_edn()
+    }
+
+    pub fn datoms(&self) -> edn::Value {
```
Comment (ncalexan) [inline db/src/debug.rs]: Can we make sure that at the end of these patches we also have `pub fn transactions(&self) ...`?  It's somewhere in these stacks of patches.

File: db/src/debug.rs [hunk 3]
```diff
@@ -414,16 +415,20 @@ impl TestConn {
         self.partition_map.get(&":db.part/tx".to_string()).unwrap().index - 1
     }
 
-    pub fn last_transaction(&self) -> edn::Value {
-        transactions_after(&self.sqlite, &self.schema, self.last_tx_id() - 1).expect("last_transaction").0[0].into_edn()
+    pub fn last_transaction(&self) -> Datoms {
+        transactions_after(&self.sqlite, &self.schema, self.last_tx_id() - 1).expect("last_transaction").0.pop().unwrap()
+    }
+
+    pub fn transactions(&self) -> Transactions {
```
Comment (ncalexan) [inline db/src/debug.rs]: Ah, there it is!

File: tolstoy/src/syncer.rs [hunk 4]
```diff
@@ -413,13 +413,86 @@ impl RemoteClient {
...
     #[test]
     fn test_remote_client_bound_uri() {
         let user_uuid = Uuid::from_str(&"316ea470-ce35-4adf-9c61-e0de6e289c59").expect("uuid");
         let server_uri = String::from("https://example.com/api/0.1");
         let remote_client = RemoteClient::new(server_uri, user_uuid);
         assert_eq!("https://example.com/api/0.1/316ea470-ce35-4adf-9c61-e0de6e289c59", remote_client.bound_base_uri());
     }
+
+    #[test]
+    fn test_add() {
```
Comment (ncalexan) [inline tolstoy/src/syncer.rs]: Maybe slim this down or do something a little more Tolstoy-esque for the test?  Your call, since it's clearly temporary.

Comment (ncalexan) [reply tolstoy/src/syncer.rs]: Maybe make one of your data types have `to_edn` and use `assert_matches!` on it?

Comment (grigoryk) [reply tolstoy/src/syncer.rs]: I will as I start landing other things. This is just to make sure everything glues together.

**Pull request merged**